archaius-core/src/main/java/com/netflix/config/DynamicBooleanProperty.java

```diff
@@ -23,14 +23,41 @@
  *
  */
 public class DynamicBooleanProperty extends PropertyWrapper<Boolean> {
+
+    protected volatile boolean primitiveValue;
+
     public DynamicBooleanProperty(String propName, boolean defaultValue) {
         super(propName, Boolean.valueOf(defaultValue));
+
+        // Set the initial value of the cached primitive value.
+        this.primitiveValue = chooseValue();
+
+        // Add a callback to update the cached primitive value when the property is changed.
+        this.prop.addCallback(new Runnable() {
```

Comment:
Rather than adding this callback here, you could just override either of the propertyChanged methods and then update the PropertyWrapper class to not add this class to the SUBCLASSES_WITH_NO_CALLBACK map.  This would be cleaner and probably result in less overall code.